File: Backend/models/zalo.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, JSON, DateTime, ForeignKey, Text, Boolean
from config.database import Base

class ZaloBot(Base):
    __tablename__ = "zalo_bot"
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    bot_name = Column(String(255), nullable=False)
    access_token = Column(String(), nullable=False, unique=True)
    description = Column(Text, nullable=True)
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.now)
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)
    company_id = Column(Integer, ForeignKey("company.id"), nullable=False)

File: Frontend/src/types/zalo.ts
// Zalo Bot Types
export interface ZaloBot {
  id: number;
  bot_name: string;
  access_token: string;
  description?: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  company_id: number;
}

export interface ZaloBotCreateRequest {
  bot_name: string;
  access_token: string;
  description?: string;
}

export interface ZaloBotUpdateRequest {
  bot_name?: string;
  access_token?: string;
  description?: string;
}

export interface ZaloBotsResponse {
  bots: ZaloBot[];
  total: number;
}
